lib/node/nodes/weighted-centroid.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'weighted-centroid';
+
+var PARAMS = {
+    source : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    weight_column : Node.PARAM.STRING,
+    category_column : Node.PARAM.STRING
+};
+
+var WeightedCentroid  = Node.create(TYPE, PARAMS, {cache: false});
+
+module.exports = WeightedCentroid;
+module.exports.TYPE = TYPE;
+module.exports.PARAMS = PARAMS;
+
+var weightedCentroidTemplate = dot.template([
+    'SELECT the_geom , class',
```

Comment:
This is more of an aggregation , it operates on a table with say 100 points and 3 classes and produces 3 points. We could add other aggregate columns, like the number of points that centroid is generated from or for example some aggregate over a column of all the points that belong to that weighted mean but I dont think its appropriate to bring back columns from the original table